segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -46,38 +47,34 @@ public IndexAppendProcessor(ScheduledExecutorService indexSegmentUpdateExecutor,
         this.store = store;
     }
 
-    /**
-     * Appends index segment on a separate thread.
-     * @param segmentName  segment name.
-     */
-    protected void processAppend(String segmentName) {
+    protected void processAppend(IndexAppend indexAppend) {
         // Not updating index segment for transient and transaction type.
-        if (isTransientSegment(segmentName) || isTransactionSegment(segmentName)) {
+        if (isTransientSegment(indexAppend.getSegment()) || isTransactionSegment(indexAppend.getSegment())) {
             return;
         }
-        LatestItemSequentialProcessor<String> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
-        latestItemProcessor.updateItem(segmentName);
+        LatestItemSequentialProcessor<IndexAppend> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
+        latestItemProcessor.updateItem(indexAppend);
     }
 
-    private void handleIndexAppend(String segmentName) {
+    private void handleIndexAppend(IndexAppend indexAppend) {
         //TODO : Update the index segment Attribute as needed going forward. If eventcount for main segment is null then we are passing it as 0 in index segment.
         AttributeUpdateCollection attributes = AttributeUpdateCollection.from(
+                //TODO: ReplaceIfGreater
                 new AttributeUpdate(EVENT_COUNT, AttributeUpdateType.Accumulate, 1));
-        store.getStreamSegmentInfo(segmentName, TIMEOUT)
-             .thenCompose(info -> store.append(getIndexSegmentName(segmentName), getIndexAppendBuf(info.getLength(),
-                     (info.getAttributes().get(EVENT_COUNT) != null) ? info.getAttributes().get(EVENT_COUNT) : 0), attributes, TIMEOUT))
-             .whenComplete((length, ex) -> {
-                 if (ex == null) {
-                     log.trace("Index segment append successful for Segment : {} with length {} ", segmentName, length);
-                 } else {
-                     log.warn("Index segment append failed for segment {} due to ", segmentName, ex);
-                 }
-             });
+        ByteBufWrapper bufferWrapper = getIndexAppendBuf(indexAppend.getOffsetLength(), indexAppend.getEventCount()); // main event count
+        // TODO: verify if this check is required or not
+        Preconditions.checkArgument( bufferWrapper.getLength() == indexAppend.getMaxEventSize(), "The data received for index segment append is not of desired size");
```

Comment:
Do we need this check here ?

Reply:
It shouldn't be here. If these sizes were different then `processAppend` above should have returned early before the code even reached here.